source/blood/src/gamemenu.cpp

```diff
@@ -3243,7 +3257,17 @@ bool CGameMenuFileSelect::Select(void)
     if (!findhigh[currentList])
         return false;
 
-    Bstrcat(destination, findhigh[currentList]->name);
+    char name[MAX_PATH];
```

Comment:
What's point of this change?

Reply:
To fix the case when user goes back from difficulty selection to user map menu and then selecting a different map. The "file browser control" remembers the last selected destination so it is needed to update it correctly, not just concatenating the newly selected file. Without the change it would be `something.mapsomething.map` instead of just `something.map`.